### DISCUSSION TITLE
Difference between envoy.ProxyConfig and meshconfig.ProxyConfig

### DISCUSSION THREAD

**Discussion by choigonyok** [istio/istio#48596] [Q&A] [answered]
Hi, I’m studying Istio codebase myself.

newProxyCommand() in /istio/pilot/cmd/pilot-agent/app/cmd.go, there are envoy.ProxyConfig and meshconfig.ProxyConfig.

What is the difference between them? help me plz :sob:

**Comment by husnialhamdani** [comment, marked as answer]
Hi,

ProxyConfig  is an option to configure per-workload basis, a per-namespace basis, or  even mesh-wide. whereas, MeshConfig  is mesh-wide setting, just like the name.

Both are also written in these documentation:
https://istio.io/latest/docs/reference/config/networking/proxy-config/
https://istio.io/latest/docs/reference/config/istio.mesh.v1alpha1/

**Reply by choigonyok** [reply]
I clearly understood. Thanks for your reply! :)
